### PR TITLE
add options for subitles in legend of cohorttrackers barplot

### LINKED ISSUE #687: Legend subtitles for CohortTracker's plot_cohort_barplot
### Description of feature

If multiple variables are tracked, or variables have many categories, the plot can become crowded quickly:
That is, the list on the right becomes potentially fairly long:

image

The option to add a subtitle in the legend could help to distinguish the different variables which are tracked.

### PR DESCRIPTION
**Pull request by eroell**
**PR Checklist**

-   [x] This comment contains a description of changes (with reason)
-   [x] Referenced issue is linked: Fixes #687 
-   [x] If you've fixed a bug or added code that should be tested, add tests!
-   [x] Documentation in `docs` is updated

**Description of changes & Technical details**
Add options for subtitles in legend of CohortTracker's `plot_cohort_barplot`:
- `legend_subtitles`: If `True`, subtitles will be added to the legend. Default is `False`.
- `legend_subtitles_names`: Dictionary to rename the legend subtitles. If `None`, the original labels will be used. The keys should be the column names.
- `fontsize`: If `True`, the subtitles will be bold. Default is `True`. NEW

**Additional context: Examples**

```py
import ehrapy as ep

adata = ep.dt.diabetes_130_fairlearn(
    columns_obs_only=["gender", "race", "num_procedures"]
)
cohort_tracker = ep.tl.CohortTracker(adata, categorical=["gender", "race"])
cohort_tracker(adata, "Initial Cohort")
adata = adata[:1000]
cohort_tracker(adata, "Filtered Cohort")
fig, ax = cohort_tracker.plot_cohort_barplot(
    subfigure_title=True,
    color_palette="tab20",
    fontsize=14,
    yticks_labels={
        "race": "Race [%]",
        "gender": "Gender [%]",
        "num_procedures": "#Procedures [mean (stdev)]",
    },
    legend_labels={
        "Unknown/Invalid": "Unknown",
        "num_procedures": "#Procedures",
    },
    legend_subtitles=True,
    legend_subtitles_names={
        "num_procedures": "Procedures",
        "race": "Race",
        "gender": "Gender",
    },
    show=False,
    legend_kwargs={"bbox_to_anchor": (1.1, 1)},
)
# can be used to adjust the position of the subplots
fig.subplots_adjust(top=1.4)
```
image

### CODE REVIEW

**Comment by eroell** [comment]
Aside from any feedback & checks from your side: what do you think about the defaults of the new arguments, @Zethson ?

File: ehrapy/tools/cohort_tracking/_cohort_tracker.py [hunk 1]
```diff
@@ -336,13 +374,49 @@ def plot_cohort_barplot(
...
         # making the plot much more readable
         legend_handles.reverse()
-        legend_handels = [item for sublist in legend_handles for item in sublist]
 
         tot_legend_kwargs = {"loc": "best", "bbox_to_anchor": (1, 1)}
         if legend_kwargs is not None:
             tot_legend_kwargs.update(legend_kwargs)
 
-        plt.legend(handles=legend_handels, **tot_legend_kwargs)
+        def create_legend_with_subtitles(patches_list, subtitles_list, tot_legend_kwargs, legend_subtitles_bold=True):
```
Comment (eroell) [inline ehrapy/tools/cohort_tracking/_cohort_tracker.py:382]: this expects such a specific structure I thought its better to not put it out of the method, to not tempt anyone to re-use this in another context^^

**Comment by Zethson** [comment]
> Aside from any feedback & checks from your side: what do you think about the defaults of the new arguments, @Zethson ?

Good!

**Comment by Zethson** [comment]
Great idea!

A few questions/requests:

1.  Is it possible to position the legend at the top next to the plot? In your example that you show here, it starts at the second bar plots which makes it look weird
2. Is there a way and/or should we have a blank line after every legend section? The bold already helps a lot.
3. I'd always make the header bold. If not, it's really hard to discern the sections. We don't need that configurable.
4. Could you please add the final (bold) plot to the reviewer response and just replace the one that we currently have?

**Pull request merged**